app/api/auth.py

```diff
@@ -146,6 +146,21 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def set_api_key(func):
```

Comment:
This function seems to replicate a lot of the logic from above. Is there any way we can make this a little more DRY?

Reply:
modified the `authenticate` decorator for resource retrieval and removed `set_api_key`